docs/sh_creation.rst

```diff
@@ -121,23 +133,23 @@ This kind of system must implement the `create_client_proxy` method:
         const YAML::Node& configuration);
 
 `soss` will call this method in order to create a new `ServiceClient` to the service `service_name`
-using the `service_type` type, and optional `configuration`. This `ServiceClient` will be provided
-as argument on the `callback` invocation when a response is received.
+using the `service_type` type, plus an optional `configuration`. This `ServiceClient` will be provided
+as an argument in the `callback` invocation when a response is received.
 
 The `ServiceClient` is an interface that must be implemented by a `Client` in order to allow `soss`
 to relate a *request* with its *reply*. This is done by providing a `call_handle` both in the
 `call_service` method from `ServiceProvider` and in the `callback` from `create_client_proxy` method.
-When the *reply* is received by another **System-Handle** its `ServiceProvider` will call the
-`received_response` metohd from the `Client`:
+When the *reply* is received by another **System-Handle**, its `ServiceProvider` will call the
+`receive_response` metohd from the `Client`:
 
 .. code-block:: cpp
 
     void receive_response(
         std::shared_ptr<void> call_handle,
         const xtypes::DynamicData& response);
 
-This `receive_response` method must translate the `response` from `service_type` and relate the
-`call_handle` if needed to its middleware's request, and reply to its middleware.
+This `receive_response` translates the `response` from `service_type`, relate the
```

Comment:
This behavior must be implemented by the user when creates the System-Handle, and I think is clearer to separate the process to implement into two steps instead of three:

- Translate the message AND relate the call_handle (if needed).
- Reply to the middleware.

This is, manage the incoming message, and manage the reply to the middleware.

Reply:
ok.

Reply:
I'll modify the entry phrase as:

    The `receive_response` must:

Because the developer of the SH must achieve that.

Reply:
Ok.